### PR TITLE
Add renaming of edge types

### PR DESCRIPTION
**Pull request by imilinovic**
There was a need to be able to change edge type so it was implemented.

[master < Epic] PR
- [ ] Check, and update documentation if necessary
- [x] Write E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message

[master < Task] PR
- [ ] Check, and update documentation if necessary
- [ ] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [x] Write a release note here, including added/changed clauses
Enable changing edge (relationship) type through the CPP (mgp::Graph::ChangeType) and C (mgp::graph_edge_change_type) API
- [x] Tag someone from docs team in the comments

### CODE REVIEW

**Comment by gitbuda** [comment]
@antoniofilipovic & @imilinovic, if I recall correctly, by definition, edge types are immutable in the property graph -> take a look for incompatibilities and edge cases on that side (maybe some application code relies on that fact) 🤔

**Comment by antoniofilipovic** [comment]
@gitbuda Basically this PR is used only for APOC functionality of renaming edge type: https://neo4j.com/labs/apoc/4.3/overview/apoc.refactor/apoc.refactor.rename.type/. Neo4j supports it so we wanted to add support on MAGE.

File: include/_mgp.hpp [hunk 1]
```diff
@@ -265,6 +265,11 @@ inline mgp_edge *graph_edge_set_to(struct mgp_graph *graph, struct mgp_edge *e,
   return MgInvoke<mgp_edge *>(mgp_graph_edge_set_to, graph, e, new_to, memory);
 }
 
+inline mgp_edge *graph_edge_change_type(struct mgp_graph *graph, struct mgp_edge *e, struct mgp_edge_type new_type,
+                                        mgp_memory *memory) {
+  return MgInvoke<mgp_edge *>(mgp_graph_edge_change_type, graph, e, new_type, memory);
+}
```
Comment (antoniofilipovic) [inline include/_mgp.hpp:271]: Why does this function return new mgp_edge? Does it to be `void` so it doesn't create new edge?

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few comments, but good work

**Review by antoniofilipovic — CHANGES_REQUESTED** [review]
Few comments, but good work

File: include/_mgp.hpp (hunk 1 above)
Comment (imilinovic) [reply include/_mgp.hpp:271]: As we discussed on slack

**Review by antoniofilipovic — APPROVED** [review]
Looks good to me

**Comment by imilinovic** [comment]
@vpavicic Will do docs today

**Pull request merged**